pkg/virt-api/rest/subresource_test.go

```diff
@@ -1167,23 +1168,75 @@ var _ = Describe("VirtualMachineInstance Subresources", func() {
 
 			statusErr := ExpectStatusErrorWithCode(recorder, http.StatusConflict)
 			// check the msg string that would be presented to virtctl output
-			Expect(statusErr.Error()).To(ContainSubstring("Halted does not support manual stop requests"))
+			Expect(statusErr.Error()).To(ContainSubstring("Halted only supports manual stop requests with a shorter graceperiod"))
 		})
 
-		It("should fail on VM with RunStrategyHalted", func() {
+		DescribeTable("should fail on VM with RunStrategyHalted", func(graceperiod *int64) {
 			vm := newVirtualMachineWithRunStrategy(v1.RunStrategyHalted)
 			vmi := newVirtualMachineInstanceInPhase(v1.Running)
 
+			var terminationGracePeriodSeconds int64 = 1800
+			vmi.Spec.TerminationGracePeriodSeconds = &terminationGracePeriodSeconds
+
+			stopOptions := &v1.StopOptions{GracePeriod: graceperiod}
+
+			bytesRepresentation, _ := json.Marshal(stopOptions)
+			request.Request.Body = io.NopCloser(bytes.NewReader(bytesRepresentation))
+
 			vmClient.EXPECT().Get(vm.Name, &k8smetav1.GetOptions{}).Return(vm, nil)
 			vmiClient.EXPECT().Get(vm.Name, &k8smetav1.GetOptions{}).Return(vmi, nil)
 
+			if graceperiod != nil {
+				vmiClient.EXPECT().Patch(vmi.Name, types.MergePatchType, gomock.Any(), gomock.Any()).DoAndReturn(
+					func(name string, patchType types.PatchType, body interface{}, opts *k8smetav1.PatchOptions) (interface{}, interface{}) {
+						//check that dryRun option has been propagated to patch request
+						Expect(opts.DryRun).To(BeEquivalentTo(stopOptions.DryRun))
+						return vm, nil
+					})
+			}
 			app.StopVMRequestHandler(request, response)
 
 			statusErr := ExpectStatusErrorWithCode(recorder, http.StatusConflict)
 			// check the msg string that would be presented to virtctl output
-			Expect(statusErr.Error()).To(ContainSubstring("Halted does not support manual stop requests"))
-		})
+			Expect(statusErr.Error()).To(ContainSubstring("Halted only supports manual stop requests with a shorter graceperiod"))
+		},
+			Entry("with nil graceperiod", nil),
+			Entry("with equal graceperiod", pointer.Int64(int64(1800))),
+			Entry("with greater graceperiod", pointer.Int64(int64(2400))),
+		)
+
+		DescribeTable("should not fail on VM with RunStrategyHalted and a shorter grace period", func(terminationGracePeriod *int64) {
```

Comment:
Since the logic is duplicated here, I'd suggest to combine it with the previous table and maybe add a parameter that indicates if we expect it to fail or not